src/StaticWebAssetsSdk/Tasks/Data/StaticWebAssetTokenResolver.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.StaticWebAssets.Tasks;
+
+public class StaticWebAssetTokenResolver(IReadOnlyDictionary<string, string> tokens)
+{
+    public static readonly StaticWebAssetTokenResolver Instance = new();
+
+    private static readonly Dictionary<string, string> _empty = [];
+
+    public StaticWebAssetTokenResolver() : this(_empty) { }
+
+    internal virtual bool TryGetValue(StaticWebAsset asset, string key, out string value)
+    {
+        if (string.Equals(key, nameof(StaticWebAsset.Fingerprint), StringComparison.OrdinalIgnoreCase))
```

Comment:
Yep, we don't expect to change fingerprint.

The way I see this is that we can replace tokens from the asset or from the environment. Tokens from the assets match their associated properties and tokens from the environment can be defined in the future.

The reason for this is that we have files like blazor.webassembly.js that we might want to stamp with a version too, like blazor.8.0.5.js so that they never get cached even in the cases where we are self-hosted (while remaining human readable).